R/mfl_scoringhistory.R

```diff
@@ -0,0 +1,61 @@
+## ff_scoringhistory (MFL) ##
+
+#' Get a dataframe of scoring history, utilizing the ff_scoring and load_player_stats functions.
+#'
+#' @param conn a conn object created by \code{ff_connect()}
+#' @param season the season of interest back to 1999
+#' @param ... other arguments
+#'
+#' @examples
+#' \donttest{
+# ssb_conn <- ff_connect(platform = "mfl", league_id = 54040, season = 2020)
+# ff_scoringhistory(ssb_conn)
+#' }
+#'
+#' @seealso \url{http://www03.myfantasyleague.com/2020/scoring_rules#rules}
+#'
+#' @describeIn ff_scoringhistory MFL: returns scoring history in a flat table, one row per player per week.
+#'
+#' @export
+ff_scoringhistory.mfl_conn <- function(conn, season, ...) {
+
+  #Pull in scoring rules for that league
+  league_rules <-
+    ff_scoring(conn) %>%
+    tidyr::separate(col = range,
+                    into = c("lower_range","upper_range"),
+                    sep = "-(?=[0-9]*$)") %>%
+    dplyr::mutate(dplyr::across(.cols = c(lower_range, upper_range),
+                                .fns = as.numeric))
+
+  #Pull Rosters from nflfastr to get positions
+  fastr_rosters <-
+    nflfastR::fast_scraper_roster(season) %>%
+    dplyr::mutate(position = dplyr::if_else(position == "FB", "RB", position))
+
+  #Load stats from nflfastr and map the rules from the internal stat_mapping file
+  fastr_games <-
+    nflfastR::load_player_stats() %>%
+    dplyr::inner_join(fastr_rosters, by = c("player_id" = "gsis_id", "season" = "season")) %>%
+    tidyr::pivot_longer(names_to = "metric",
+                        cols = c(completions, attempts, passing_yards, passing_tds, interceptions, sacks, sack_fumbles_lost,
```

Comment:
These should be strings, I think! Datapasta should make quick work of converting it